packages/paste-core/components/tooltip/src/Tooltip.tsx

```diff
@@ -60,42 +62,111 @@ export interface TooltipProps extends TooltipPrimitiveInitialState {
    */
   text: string;
+  actionHeader?: never;
+  keyCombinationsActions?: never;
 }
 
+interface KeyboardActions {
+  name: string;
+  eventKeyCombination: string[];
+  disabled?: boolean;
+}
+
+export interface KeyboardKeyTooltipProps
+  extends Omit<TooltipProps, "text" | "keyCombinationsActions" | "actionHeader"> {
+  text?: never;
+  /**
+   * The mapping of action names to their respective key combinations.
+   *
+   * @type {Array<KeyboardActions>}
+   * @memberof KeyboardKeyTooltipProps
+   */
+  keyCombinationsActions: Array<KeyboardActions>;
+  /**
+   * The header content of the Tooltip.
+   *
+   * @type {string}
+   * @memberof KeyboardKeyTooltipProps
+   */
+  actionHeader?: string;
+}
+
+// Union will stop users from adding types from both TooltipProps and KeyboardKeyTooltipProps at the same time.
+export type TooltipVariantProps = TooltipProps | KeyboardKeyTooltipProps;
+
 /*
  *Tooltip's current structure does not allow for customization of its arrow.
  *TODO: refactor Tooltip so that the styling of its arrow can be customized
  *using Customization Provider.
  */
 
-const Tooltip = React.forwardRef<HTMLDivElement, TooltipProps>(
-  ({ baseId, children, element = "TOOLTIP", state, text, ...props }, ref) => {
+const Tooltip = React.forwardRef<HTMLDivElement, TooltipVariantProps>(
+  ({ baseId, children, element = "TOOLTIP", state, text, actionHeader, keyCombinationsActions, ...props }, ref) => {
     const tooltip = state || useTooltipPrimitiveState({ baseId: `paste-tooltip-${useUID()}`, ...props });
+
     return (
       <>
         {React.Children.only(
           <TooltipPrimitiveReference {...tooltip} ref={ref} {...children.props}>
             {(referenceProps) => React.cloneElement(children, referenceProps)}
-          </TooltipPrimitiveReference>,
+          </TooltipPrimitiveReference>
         )}
         <TooltipPrimitive element={element} {...tooltip} {...props} as={StyledTooltip}>
           {/* import Paste Theme Based Styles due to portal positioning. */}
           <StyledBase>
             <TooltipArrow {...tooltip} />
-            <Text
-              element={`${element}_TEXT`}
-              as="span"
-              color="colorTextInverse"
-              fontSize="fontSize20"
-              lineHeight="lineHeight10"
-            >
-              {text}
-            </Text>
+            {text && !keyCombinationsActions && (
+              <Text
+                element={`${element}_TEXT`}
+                as="span"
+                color="colorTextInverse"
+                fontSize="fontSize20"
+                lineHeight="lineHeight10"
+              >
+                {text}
+              </Text>
+            )}
+            {keyCombinationsActions && (
+              <Stack orientation="vertical" spacing="space30">
+                {actionHeader && (
+                  <Text element={`${element}_HEADER`} fontWeight="fontWeightSemibold" as="p" color="colorTextInverse">
+                    {actionHeader}
```

Comment:
todo: text in header should also be lineHeight10 and fontSize20

Reply:
done!